payments/src/lib.rs

```diff
@@ -583,10 +597,20 @@ pub mod pallet {
 
 					// Calculate fee amount - this will be implemented based on the custom
 					// implementation of the fee provider
-					let (fee_recipient, fee_percent) = T::FeeHandler::apply_fees(from, recipient, &new_payment, remark);
-					let fee_amount = fee_percent.mul_floor(amount);
-					new_payment.fee_detail = Some((fee_recipient, fee_amount));
+					let recipients = T::FeeHandler::apply_fees(from, recipient, &new_payment, remark);
 
+					let mut fee_recipient_list: FeeRecipientList<T> = Default::default();
+
+					for fee_recipient in recipients {
```

Comment:
Maybe a `.map` could work better here?  `fee_recipient_list` has the same length as `recipients` right? then the `FeeCalculationFailed` error can be avoided since it's misleading anyway as it shouldn't happen.